datalad/distribution/create_publication_target_sshwebserver.py

```diff
@@ -284,13 +285,28 @@ def __call__(sshurl, target=None, target_dir=None,
                             "You will not be able to push to checked out "
                             "branch." % path)
 
-            # enable post-update hook:
+            # create post_update script
+            lgr.info("Enabling git post-update hook ...")
             try:
-                ssh(["mv",
-                     opj(path, ".git/hooks/post-update.sample"),
-                     opj(path, ".git/hooks/post-update")])
+                json_command = '"datalad ls -r --json file ' + path + '"'
+                virtualenv = 'source /home/debanjum/datalad/.env/bin/activate'  # NOT_SCALABLE! custom virtualenv path
+                for post_update_hook in ['"#!/bin/bash"', '"git update-server-info"', virtualenv, json_command]:
+                    ssh(["echo", post_update_hook, '>>', opj(path, '.git/hooks/post-update')])
+                ssh(["chmod", '+x', opj(path, '.git/hooks/post-update')])  # make the script executable
             except CommandError as e:
-                lgr.error("Failed to enable post update hook.\n"
+                lgr.error("Failed to add json creation command to post update hook.\n"
+                          "Error: %s" % e.message)
```

Comment:
Generally better to use `exc_str` helper while logging information about caught exception, since then we could control of how much information (traceback) is provided within the message -- came really handy few times